src/main/java/com/khomishchak/cryptoportfolio/controllers/ExchangerController.java

```diff
@@ -46,22 +48,25 @@ public List<Balance> getAccountBalances(@RequestAttribute long userId) {
         return exchangerService.getAllMainBalances(userId);
     }
 
-    @DeleteMapping("/{exchangerCode}/balance")
-    public DeleteExchangerResp deleteAccountBalance(@RequestAttribute long userId, @PathVariable String exchangerCode) {
-        return exchangerService.deleteExchangerForUser(userId, ExchangerCode.valueOf(exchangerCode));
+    @DeleteMapping("/{balanceId}/balance")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteAccountBalance(@PathVariable long balanceId) {
+        exchangerService.deleteExchangerForUser(balanceId);
     }
 
-    @GetMapping("/{exchangerCode}/wallet/deposit-withdrawal-history")
-    public List<DepositWithdrawalTransaction> getAccountWithdrawalDepositWalletHistory(@RequestAttribute long userId,
-            @PathVariable String exchangerCode) {
-        return exchangerService.getWithdrawalDepositWalletHistory(userId, ExchangerCode.valueOf(exchangerCode));
+    @GetMapping("/deposit-withdrawal-history")
```

Comment:
Same as above. From reader perspective and constructing an API this endpoint returns deposit withdrawal history of all exchangers for some user id. Is that right?

Reply:
yes, that's right, because this history is also related to each of the exchangers integrations. there no sense to create a new controller with just 2-3 methods related to deposit withdrawal history